network/src/protocols/ping.rs

```diff
@@ -216,36 +210,37 @@ impl ServiceProtocol for PingHandler {
                         } else {
                             ps.processing = true;
                             ps.last_ping = now;
-                            Some((*session_id, ps.nonce()))
+                            Some(*session_id)
                         }
                     })
                     .collect();
                 if !peers.is_empty() {
-                    let ping_msg = PingMessage::build_ping(peers[0].1);
-                    let peer_ids: Vec<SessionId> = peers
-                        .into_iter()
-                        .map(|(session_id, _)| session_id)
-                        .collect();
+                    debug!("start ping peers: {:?}", peers);
+                    let ping_msg = PingMessage::build_ping(nonce(&now));
                     let proto_id = context.proto_id;
                     if context
-                        .filter_broadcast(TargetSession::Multi(peer_ids), proto_id, ping_msg)
+                        .filter_broadcast(TargetSession::Multi(peers), proto_id, ping_msg)
                         .is_err()
                     {
                         debug!("send message fail");
                     }
                 }
             }
             CHECK_TIMEOUT_TOKEN => {
-                debug!("proto [{}] check ping timeout", context.proto_id);
                 let timeout = self.timeout;
-                for peer_id in self
-                    .connected_session_ids
-                    .values()
-                    .filter(|ps| ps.processing && ps.elapsed() >= timeout)
-                    .map(|ps| ps.peer_id.clone())
-                    .collect::<Vec<PeerId>>()
-                {
-                    self.send_event(Event::Timeout(peer_id));
+                for id in self.connected_session_ids.iter().filter_map(|(id, ps)| {
+                    if ps.processing && ps.elapsed() >= timeout {
+                        Some(id)
+                    } else {
+                        None
+                    }
+                }) {
```

Comment:
```suggestion
for (id, _ps) in self
    .connected_session_ids
    .iter()
    .filter(|(_id, ps)| ps.processing && ps.elapsed() >= timeout)
{
}
```

Reply:
This is just a little bit of code reduction, the essence has not changed, I have no opinion on this

Reply:
:+1: for the suggested change

Reply:
done